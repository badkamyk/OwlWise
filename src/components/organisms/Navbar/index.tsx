import { useState } from "react";
import NavButton from "../../atoms/NavButton";
import NavLogo from "../../atoms/NavLogo";
import AllNavLinks from "../../molecules/AllNavLinks";

export default function Navbar() {
	const [isExpanded, setIsExpanded] = useState(false);

	return (
		<nav className="border-gray-200 bg-white px-2 py-2.5 dark:bg-gray-900 sm:px-4">
			<div className="container mx-auto flex flex-wrap items-center justify-between">
				<NavLogo
					href="https://flowbite.com/"
					src="https://flowbite.com/docs/images/logo.svg"
					alt="Flowbite"
					text="OwlWise"
				/>
				<NavButton isExpanded={isExpanded} setIsExpanded={setIsExpanded} />
				<AllNavLinks isExpanded={isExpanded} />
			</div>
		</nav>
	);
}
