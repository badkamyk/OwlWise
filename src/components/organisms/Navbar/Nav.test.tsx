import { render, screen } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import Navbar from "./index";

const navLinks = ["Home", "About", "Pricing", "Contact"];
describe("Navbar", () => {
	it("renders Navbar correctly with NavLogo, NavButton and AllNavLinks", () => {
		const { container } = render(<Navbar />);
		expect(
			container.getElementsByClassName(
				"border-gray-200 bg-white px-2 py-2.5 dark:bg-gray-900 sm:px-4"
			)
		).toBeTruthy();
		expect(screen.getByTestId("nav-logo")).toBeTruthy();
		expect(screen.getByTestId("nav-button")).toBeTruthy();
		expect(screen.getByTestId("all-nav-links")).toBeTruthy();
		expect(screen.getAllByTestId("nav-link")).toHaveLength(navLinks.length);
	});
});
