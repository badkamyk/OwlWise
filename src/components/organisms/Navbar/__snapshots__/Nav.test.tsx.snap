// Vitest Snapshot v1

exports[`Navbar > renders Navbar correctly with NavLogo, NavButton and AllNavLinks 1`] = `
<div>
  <nav
    class="border-gray-200 bg-white px-2 py-2.5 dark:bg-gray-900 sm:px-4"
  >
    <div
      class="container mx-auto flex flex-wrap items-center justify-between"
    >
      <a
        class="flex items-center"
        data-testid="nav-logo"
        href="/"
      >
        <img
          alt="OwlWise"
          class="mr-3 h-6 sm:h-9"
          src="https://flowbite.com/docs/images/logo.svg"
        />
        <span
          class="self-center whitespace-nowrap text-xl font-semibold dark:text-white"
        >
          OwlWise
        </span>
      </a>
      <button
        aria-controls="navbar-default"
        aria-expanded="false"
        class="ml-3 inline-flex items-center rounded-lg p-2 text-sm text-gray-500 hover:bg-gray-100 focus:outline-none focus:ring-2 focus:ring-gray-200 dark:text-gray-400 dark:hover:bg-gray-700 dark:focus:ring-gray-600 md:hidden"
        data-collapse-toggle="navbar-default"
        data-testid="nav-button"
        type="button"
      >
        <span
          class="sr-only"
        >
          Open main menu
        </span>
        <svg
          aria-hidden="true"
          class="h-6 w-6"
          fill="currentColor"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M3 5a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 10a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 15a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1z"
            fill-rule="evenodd"
          />
        </svg>
      </button>
      <div
        class="hidden w-full md:block md:w-auto"
        data-testid="all-nav-links"
        id="navbar-default"
      >
        <ul
          class="mt-4 flex flex-col rounded-lg border border-gray-100 bg-gray-50 p-4 dark:border-gray-700 dark:bg-gray-800 md:mt-0 md:flex-row md:space-x-8 md:border-0 md:bg-white md:text-sm md:font-medium md:dark:bg-gray-900"
        >
          <li>
            <a
              aria-current="page"
              class="block rounded bg-blue-700 py-2 pl-3 pr-4 text-white dark:text-white md:bg-transparent md:p-0 md:text-blue-400"
              data-testid="nav-link"
              href="/"
            >
              Home
            </a>
          </li>
          <li>
            <a
              aria-current="page"
              class="block rounded bg-blue-700 py-2 pl-3 pr-4 text-white dark:text-white md:bg-transparent md:p-0 md:text-blue-400"
              data-testid="nav-link"
              href="/about"
            >
              About
            </a>
          </li>
          <li>
            <a
              aria-current="page"
              class="block rounded bg-blue-700 py-2 pl-3 pr-4 text-white dark:text-white md:bg-transparent md:p-0 md:text-blue-400"
              data-testid="nav-link"
              href="/pricing"
            >
              Pricing
            </a>
          </li>
          <li>
            <a
              aria-current="page"
              class="block rounded bg-blue-700 py-2 pl-3 pr-4 text-white dark:text-white md:bg-transparent md:p-0 md:text-blue-400"
              data-testid="nav-link"
              href="/contact"
            >
              Contact
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
</div>
`;
